src/gui_common/charts/DataPoint.cs

```diff
@@ -103,6 +106,43 @@ public float Size
         return !(lhs == rhs);
     }
 
+    /// <summary>
+    ///   Get a visible datapoint from the cache or a new one if the cache is empty. Due to the fact that this isn't a
+    ///   constructor, all common parameters that needs to be initialized can be passed passed in as optional
+    ///   parameters.
+    /// </summary>
+    public static DataPoint GetDataPoint(double x, double y, MarkerIcon iconType = MarkerIcon.Circle, float size = 7,
+        Color markerColour = default, bool draw = true, Vector2 coordinate = default)
+    {
+        if (DataPointCache.Count == 0)
+        {
+            return new DataPoint(x, y)
+                { IconType = iconType, Size = size, MarkerColour = markerColour, Draw = draw, coordinate = coordinate };
+        }
+
+        var point = DataPointCache.Pop();
+        point.X = x;
+        point.Y = y;
+        point.IconType = iconType;
+        point.Size = size;
+        point.MarkerColour = markerColour;
+        point.Draw = draw;
+        point.coordinate = coordinate;
+        point.Visible = true;
+        return point;
+    }
+
+    /// <summary>
+    ///   Returns a datapoint to the cache. Make sure that the point isn't used anywhere else before returning it to
+    ///   the cache.
+    /// </summary>
+    public static void RestoreDataPoint(DataPoint point)
```

Comment:
```suggestion
    public static void ReturnDataPoint(DataPoint point)
```
I think the method name would be clearer that way and would also match the documentation text.